### ISSUE TITLE
Sorting doesn't work with Wikilinks

### ISSUE THREAD

**Issue by janpeeters70** [Vinzent03/obsidian-sort-and-permute-lines#2]
Thanks for your very useful plugin.
I noticed today that sorting of lists of wikilinks
[[Name A]]
[[Source B]]
[[Note C]]

Doesn't work. Would come in handy for TOC's and MOC's.

Thanks, Jan

**Comment by janpeeters70** [comment]
Hi @Vinzent03, yes I'm on the latest version. What's happening is that I create a list of wikilinks, select them and chose the Sort Aphabetically option from the command window. It does do something just not sort it in the right order.
I have included a video in which you can see the behavior. 
At the end I undo and redo the sorting so that you can see what changed. 

https://user-images.githubusercontent.com/4502360/114988229-61eae880-9e96-11eb-9881-d84f2e74d484.mp4

**Comment by Vinzent03** [comment]
I am really confused. I cannot reproduce it with the same text. (Windows and Linux, although I don't know why the OS should make a difference). In my case it sort correctly from
```
[[So that its a real list]]
[[This is another note]]
[[More notes]]
[[Almost a list now]]
[[Better type on more]]
[[A note]]
```
to
```
[[Almost a list now]]
[[A note]]
[[Better type on more]]
[[More notes]]
[[So that its a real list]]
[[This is another note]]
```
The only thing I think of is to remove white spaces before sorting. But I don't know why in your case `Bettter...`comes after `More...` . Are there any error in the console?(`ctrl + shift + i`)

**Comment by janpeeters70** [comment]
This is indeed really weird. There's no error in the console.
I've tried deactivating all community plugins and the California Coast theme but that didn't change a thing.

I've tried to remove the spaces but even then it doesn't sort the lines right. It's not that it doesn't work at all some lines look they're being sorted just one or two are out of order. I've tried to sort on line length, but it also doesn't work. 

When I remove the wikilink characters [[ the alphabetical sorting works just fine. But the sorting on line length doesn't.

So I don't know what makes it not work.

**Comment by Vinzent03** [comment]
I have not idea what's the problem. Everything is working fine for me.  Which OS are you on? Are you so interested in fixing this, that you would try an extra version with a lot of logging?

**Comment by janpeeters70** [comment]
I'm on macOS 11.2.3.
I would be interested to get this fixed so also to run a test version with lots of logging.
Let me know when you have something. This weekend I'm poorly available but from next Monday I can test.

**Comment by Vinzent03** [comment]
Sorry for the late answer. I was very busy with [Obsidian-git](https://github.com/denolehov/obsidian-git). Please replace your `<vault>/.obsidian/plugins/obsidian-sort-and-permute-lines/main.js` with the `main.js` from the attached zip. Sorry for zip, GitHub doesn't allow `.js` files. Then restart Obsidian and send me your logs. Thanks.
[main.zip](https://github.com/Vinzent03/obsidian-sort-and-permute-lines/files/6356499/main.zip)

**Comment by janpeeters70** [comment]
Hi @Vinzent03, no problem. I was also busy.

I just downloaded your zip and wanted to install it. When I opened your plugin folder I found duplicates of all files with a ' 2' after the filename.
I'm not sure what made the duplicates, maybe the beta Obsidian iOS client that I'm testing. 
After removing these files everything worked like it should.

To be honest it might as well be something else that fixed it, because I wouldn't know how these duplicates would influence the behavior of the plugin.

So it's fixed. Sorry that you had to put time in supporting me with this.

I'm going to have a look at getting Obsidian Git to work now. I seem to have been using an old authentication method when setting up my git and it stopped synching :-(
Thanks for developing these plugins! Really appreciated.
